File: data-structures/cache/ring-buffer/ringBuffer.js
class RingBuffer {
  constructor(capacity) {
    if (typeof capacity !== 'number') return {};
    this.capacity = capacity;
    this.current = 0;
    this.buffer = [...Array(capacity)].fill(null);
  }

  append(value) {
    this.buffer[this.current] = value;
    this.current++;
    if (this.current === this.capacity) this.current = 0;
  }

  findItem(value) {
    for (let i = 0; i < this.buffer.length; i++) {
      if (this.buffer[i] && this.buffer[i] == value) return true;
    }
    return false;
  }

  getBuffer() {
    const final = [];
    for (let i = 0; i < this.buffer.length; i++) {
      if (this.buffer[i]) final.push(this.buffer[i]);
    }
    return final;
  }
}

module.exports = RingBuffer;

File: data_structures/cache/ring_buffer/ring_buffer.py
class Ring_Buffer:
    def __init__(self, capacity):
        self.capacity = capacity
        self.current = 0
        self.buffer = [None] * capacity

    def append(self, value):
        self.buffer[self.current] = value
        self.current += 1
        if self.current == self.capacity:
            self.current = 0

    def contains_item(self, value):
        contains_item = False
        for item in self.buffer:
            if item == value:
                contains_item = True
        return contains_item

    def get_buffer(self):
        return [i for i in self.buffer if i is not None]
